lib/patientFinder/findPatientsIDs.rb

```diff
@@ -0,0 +1,203 @@
+		end
+	end
+
+
+end
+
+
+# a collection of utilities grouped together under the same roof
+class Utils
+
+	#resolves the candidates to the test files (one test file - one candidate - when multiple candidates are found manual anotation is used)
+	def self.getUniqueMedicalRecordNumber(candidates,testRecord,testToMasterHash)
+
+
+		#currently the candidates are records - get only the IDs out of all those records
+		candidateIDs=Utils::collectMedicalRecordNumbers(candidates)
+
+		if(candidateIDs.length==1)
+			testToMasterHash[testRecord.fileName]=candidateIDs.first			
+		elsif((candidateIDs.length==2) & (candidateIDs & ["ee85c89f24946e2ddca12c6edc5181dc", "eb6af018e11320f5c163624beb83767d"]).present?)
+			testToMasterHash[testRecord.fileName]="ee85c89f24946e2ddca12c6edc5181dc"			
+		elsif((candidateIDs.length==4) & (candidateIDs & ["7a86cc6d87cb84461ea190e7c706d81b", "1be81ce59da982792026cc82d95bc10e","511b530c8662f8df97eb97b3eefa0618","258ee9087c5a5fe359ceb3aafff0dd76"]).present?)
+			testToMasterHash[testRecord.fileName]="258ee9087c5a5fe359ceb3aafff0dd76"			
+		elsif((candidateIDs.length==4) & (candidateIDs & ["0a26a978d07240b0f917e96727db31d8","ee85c89f24946e2ddca12c6edc5181dc","343731b99c6bdb35a1fc31ce9ed6f889","eb6af018e11320f5c163624beb83767d"]).present?)
+			testToMasterHash[testRecord.fileName]="0a26a978d07240b0f917e96727db31d8"	
+		elsif((candidateIDs.length==2) & (candidateIDs & ["929dd1f2c4c54c024fd4d18b1307fdb1", "29cecb1da08efd331fce823e12b607d5"]).present?)
+			testToMasterHash[testRecord.fileName]="929dd1f2c4c54c024fd4d18b1307fdb1"	
+		elsif((candidateIDs.length==2) & (candidateIDs & ["5f118631f09abdbdeb1962dc28bfeb27", "a6300c43651965991a4308ffaeb5381d"]).present?)
+			testToMasterHash[testRecord.fileName]="5f118631f09abdbdeb1962dc28bfeb27"	
+		elsif((candidateIDs.length==2) & (candidateIDs & ["7a86cc6d87cb84461ea190e7c706d81b", "258ee9087c5a5fe359ceb3aafff0dd76"]).present?)
+			testToMasterHash[testRecord.fileName]="258ee9087c5a5fe359ceb3aafff0dd76"	
+		elsif((candidateIDs.length>8) & (candidateIDs & ["fed089904c10b81c036adddedddebe7b", "846c1c2ba8370c2f5504b315cc4b1d5d", "b5633133e3421216ced2bdef4dbf382d", "0a26a978d07240b0f917e96727db31d8","ee85c89f24946e2ddca12c6edc5181dc","b54a4e3ab37de7e5f8094793afb8a699","2678a4e396aaec03b860d5aeadcad8e6","929dd1f2c4c54c024fd4d18b1307fdb1","fdea0c22270417d9e59f20b07f642679","91bd37f9cebf7b6ef9f72d7fd6148a81","0085074bb549ffefffa6e16ff34df140","343731b99c6bdb35a1fc31ce9ed6f889","eb6af018e11320f5c163624beb83767d","470f57b022eaeffd4d599078e851a56d","0dbaf9336f7aa1590265250a0eebe548","29cecb1da08efd331fce823e12b607d5","d156a6d38e10efc30eda3cace7456537","8130b2ff5774f1593c86eba8dca4c37b","697e147f076648275e518e7b3ff41dcd"]).present?)
+			testToMasterHash[testRecord.fileName]="fed089904c10b81c036adddedddebe7b"	
+		else
+			for v in candidateIDs
+				puts "\t#{v}"
+			end 
+		end
+	end	
+
+	#extracts the ids from a set of records 
+	def self.collectMedicalRecordNumbers(candidates)
+		ids=Array.new
```

Comment:
This function can be replaced with `candidates.map(&:medical_record_number) `. In general you don't have to declare variables in ruby before using them (even implicitly like by setting them to be `Array.new`)